cli/internal/daemon/status.go

```diff
@@ -0,0 +1,98 @@
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/daemon/connector"
+	"github.com/vercel/turborepo/cli/internal/daemonclient"
+	"github.com/vercel/turborepo/cli/internal/fs"
+)
+
+func addStatusCmd(root *cobra.Command, config *config.Config, output cli.Ui) {
+	var outputJSON bool
+	cmd := &cobra.Command{
+		Use:           "status",
+		Short:         "Reports the status of the turbo daemon",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := &status{
+				logger:       config.Logger,
+				output:       output,
+				turboVersion: config.TurboVersion,
+				repoRoot:     config.Cwd,
+				outputJSON:   outputJSON,
+			}
+			return s.status()
+		},
+	}
+	cmd.Flags().BoolVar(&outputJSON, "json", false, "Pass --json to report status in JSON format")
+	root.AddCommand(cmd)
+}
+
+type status struct {
+	logger       hclog.Logger
+	output       cli.Ui
+	turboVersion string
+	repoRoot     fs.AbsolutePath
+	outputJSON   bool
+}
+
+func (s *status) status() error {
+	ctx := context.Background()
+	client, err := GetClient(ctx, s.repoRoot, s.logger, s.turboVersion, ClientOpts{
+		DontStart: true,
+	})
+	if err != nil {
+		return s.reportError(err)
+	}
+	turboClient := daemonclient.New(ctx, client)
+	status, err := turboClient.Status()
+	if err != nil {
+		return s.reportError(err)
+	}
+	if s.outputJSON {
+		rendered, err := json.MarshalIndent(map[string]interface{}{
+			"logFile":  status.LogFile,
```

Comment:
`sockFile`? or too likely to end up in people doing bad things?

Reply:
Nah, we should include it, and the pid file, I think. I think even better would be having `daemonclient` wrap the status so we aren't dealing with the proto object directly. That would also be the natural place to add status information from the client.

Reply:
Done.